### PR TITLE
Support data type int8_t . 

### PR DESCRIPTION
**Pull request by qingqing01**

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_tensor.py [hunk 1]
```diff
@@ -59,6 +59,26 @@ def test_float_tensor(self):
...
+            -127, high=128, size=[100, 200], dtype=numpy.int8)
+
+        place = core.CPUPlace()
+        cpu_tensor.set(tensor_array, place)
+        cpu_tensor_array_2 = numpy.array(cpu_tensor)
+        self.assertAlmostEqual(cpu_tensor_array_2.all(), tensor_array.all())
+
+        tensor_array = numpy.random.randint(
+            -127, high=128, size=[100, 200], dtype=numpy.int8)
+        place = core.CUDAPlace(0)
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/test_tensor.py]: if core.is_compiled_with_cuda()?

**Review by panyx0718 — DISMISSED** [review]

File: python/paddle/fluid/tests/unittests/test_tensor.py (hunk 1 above)
Comment (qingqing01) [reply python/paddle/fluid/tests/unittests/test_tensor.py]: Done. Thanks!

**Comment by qingqing01** [comment]
@panyx0718 @typhoonzero  need review for API.

**Review by typhoonzero — APPROVED** [review]
Approve for API

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**